src/input/keyboard/mod.rs

```diff
@@ -556,6 +556,24 @@ impl<D: SeatHandler + 'static> KeyboardHandle<D> {
         );
     }
 
+    /// Iterate over the keysyms of the currently pressed keys.
+    pub fn with_pressed_keysyms<F>(&self, mut f: F)
+    where
+        F: FnMut(KeysymHandle<'_>),
```

Comment:
Should this have a return value?

Reply:
Maybe for the case of running the closure once with an iterator, but the semantics of merging results for running this multiple times, seem to be unnecessary complex.